### DISCUSSION TITLE
Cannot install laravel-sitemap

### DISCUSSION THREAD

**Discussion by teobatman00** [spatie/laravel-sitemap#400] [Q&A] [answered]
Hello all developers, I would like to install sitemap to laravel before uploading to hosting but I got the problems that could not be resolved. Could anyone help me? 

**Terminal**
`composer require spatie/laravel-sitemap `

**Result**

> spatie/laravel-sitemap[6.0.0, ..., 6.0.2] require spatie/crawler ^5.0 || ^6.0 -> satisfiable by spatie/crawler[5.0.0, 5.0.1, 5.0.2, v5.x-dev, 6.0.0, 6.0.1, 6.0.2, v6.x-dev].
    - spatie/laravel-sitemap[6.0.3, ..., 6.0.4] require spatie/crawler ^5.0 || ^7.0 -> satisfiable by spatie/crawler[5.0.0, 5.0.1, 5.0.2, v5.x-dev, 7.0.0].
    - spatie/crawler[6.0.0, ..., v6.x-dev] require guzzlehttp/psr7 ^1.7 -> found guzzlehttp/psr7[1.7.0, ..., 1.x-dev] but the package is fixed to 2.0.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - spatie/crawler[5.0.2, ..., v5.x-dev] require guzzlehttp/psr7 ^1.4 -> found guzzlehttp/psr7[1.4.0, ..., 1.x-dev] but the package is fixed to 2.0.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - spatie/crawler[5.0.0, ..., 5.0.1] require php ^7.4 -> your php version (8.0.6) does not satisfy that requirement.
    - spatie/crawler 7.0.0 requires guzzlehttp/psr7 ^1.8 -> found guzzlehttp/psr7[1.8.0, 1.8.1, 1.8.2, 1.x-dev] but the package is fixed to 2.0.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - Root composer.json requires spatie/laravel-sitemap ^6.0 -> satisfiable by spatie/laravel-sitemap[6.0.0, ..., 6.0.4].
 
**This is all the problem**

**Comment by ousid** [comment, marked as answer]
Hey, @teobatman00 

The `laravel-sitemap` requires `php8` installed on your system

See more here:
https://github.com/spatie/laravel-sitemap/blob/master/CHANGELOG.md#600---2021-03-12

**Reply by teobatman00** [reply]
Ok, I will check it later.

**Comment by theprobugmaker** [comment]
I'm having the same problem and I have PHP 8 installed (I'm actually using it)

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - spatie/laravel-sitemap[6.0.0, ..., 6.0.2] require spatie/crawler ^5.0 || ^6.0 -> satisfiable by spatie/crawler[5.0.0, 5.0.1, 5.0.2, v5.x-dev, 6.0.0, 6.0.1, 6.0.2, v6.x-dev].
    - spatie/laravel-sitemap[6.0.3, ..., 6.0.4] require spatie/crawler ^5.0 || ^7.0 -> satisfiable by spatie/crawler[5.0.0, 5.0.1, 5.0.2, v5.x-dev, 7.0.0].
    - spatie/crawler[6.0.0, ..., v6.x-dev] require guzzlehttp/psr7 ^1.7 -> found guzzlehttp/psr7[1.7.0, ..., 1.x-dev] but the package is fixed to 2.0.0 (lock file version) by a partial update and that version does not match. Make su
re you list it as an argument for the update command.
    - spatie/crawler[5.0.2, ..., v5.x-dev] require guzzlehttp/psr7 ^1.4 -> found guzzlehttp/psr7[1.4.0, ..., 1.x-dev] but the package is fixed to 2.0.0 (lock file version) by a partial update and that version does not match. Make su
re you list it as an argument for the update command.
    - spatie/crawler[5.0.0, ..., 5.0.1] require php ^7.4 -> your php version (8.0.7) does not satisfy that requirement.
    - spatie/crawler 7.0.0 requires guzzlehttp/psr7 ^1.8 -> found guzzlehttp/psr7[1.8.0, 1.8.1, 1.8.2, 1.x-dev] but the package is fixed to 2.0.0 (lock file version) by a partial update and that version does not match. Make sure you
 list it as an argument for the update command.
    - Root composer.json requires spatie/laravel-sitemap ^6.0 -> satisfiable by spatie/laravel-sitemap[6.0.0, ..., 6.0.4].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.
```

**Reply by amilmdr** [reply]
"php": "^7.3|^8.0",
        "fruitcake/laravel-cors": "^2.0",
        "guzzlehttp/guzzle": "^7.2",
        "guzzlehttp/psr7": "^1.4",
        "illuminate/support": "^8.0",
        "laravel/framework": "^8.75",
        "laravel/sanctum": "^2.11",
        "laravel/tinker": "^2.5",
        "nesbot/carbon": "^2.0",
        "spatie/crawler": "^5.0|^6.0",
        "spatie/laravel-sitemap": "5.9.2",
        "symfony/dom-crawler": "^5.1.14"

**Reply by amilmdr** [reply]
composer update  
that  work for me

**Reply by nirob110** [reply]
Worked for me too.

**Reply by nhantrandev96** [reply]
@amilmdr god bless you

**Comment by theprobugmaker** [comment]
cc @freekmurze @ousid the answer that php8 is required was accepted but I can see that we are both using PHP 8.

> your php version (8.0.7) does not satisfy that requirement.

**Reply by ousid** [reply]
Which version are you using?
- Try to clear you cache `php artisan optimize:clear`
- remove composer.lock
- `composer update`

## Edit
Look at your all package and make sure the provided version supports php8, cause likely this error does not come from `laravel-sitemap` package, and maybe it's better to provide us with your `composer.json` file

**Reply by theprobugmaker** [reply]
@ousid 

I did the steps but I'm still geting errors. This is how my `composer.json` file looks like:

```json
{
    "name": "laravel/laravel",
    "type": "project",
    "description": "The Laravel Framework.",
    "keywords": [
        "framework",
        "laravel"
    ],
    "license": "MIT",
    "require": {
        "php": "^8.0",
        "cviebrock/eloquent-sluggable": "^8.0",
        "fideloper/proxy": "^4.4",
        "fruitcake/laravel-cors": "^2.0",
        "guzzlehttp/guzzle": "^7.0.1",
        "laravel/framework": "^8.40",
        "laravel/tinker": "^2.5",
        "mcamara/laravel-localization": "^1.6",
        "predis/predis": "^1.1",
        "rollbar/rollbar-laravel": "^7.0",
        "symfony/css-selector": "^5.3",
        "symfony/dom-crawler": "^5.3"
    },
    "require-dev": {
        "barryvdh/laravel-ide-helper": "^2.10",
        "facade/ignition": "^2.5",
        "fakerphp/faker": "^1.9.1",
        "laravel/sail": "^1.0.1",
        "mockery/mockery": "^1.4.2",
        "nunomaduro/collision": "^5.0",
        "phpunit/phpunit": "^9.3.3"
    },
    "autoload": {
        "psr-4": {
            "App\\": "app/",
            "Database\\Factories\\": "database/factories/",
            "Database\\Seeders\\": "database/seeders/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "scripts": {
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover --ansi"
        ],
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ],
        "post-create-project-cmd": [
            "@php artisan key:generate --ansi"
        ],
        "post-update-cmd": [
            "Illuminate\\Foundation\\ComposerScripts::postUpdate",
            "@ide-helper"
        ],
        "ide-helper": [
            "@php artisan clear-compiled",
            "@php artisan ide-helper:generate",
            "@php artisan ide-helper:meta",
            "@php artisan ide-helper:models -N"
        ]
    },
    "extra": {
        "laravel": {
            "dont-discover": []
        }
    },
    "config": {
        "optimize-autoloader": true,
        "preferred-install": "dist",
        "sort-packages": true
    },
    "minimum-stability": "dev",
    "prefer-stable": true
}
```

**Reply by theprobugmaker** [reply]
@ousid 

I just tried:

> composer create-project laravel/laravel example-app
> composer require spatie/laravel-sitemap

The same problem. I think it's something with the package.

I'm using PHP 8.0.8.

**Reply by ousid** [reply]
Hey, @zefexdeveloper 
Sorry for late response. This is odd, I have `php8.0.5` installed, and it works fine to me, TBH I don't test it on `8.0.8`
What's your composer version?

**Reply by theprobugmaker** [reply]
Hi @ousid , I'm using Composer 2.1.3.

**Reply by teobatman00** [reply]
I have got the same problem with @zefexdeveloper  again recently, how can I fix it ?

**Comment by mikerockett** [comment]
It seems to me like the crawler package is the problem here. It requires `guzzlehttp/psr7 ^1.8`, but Composer installs `^2.0` via `guzzlehttp/guzzle ^7.0.1`, which is a Laravel requirement.

cc @freekmurze

**Reply by nguyenmiinh0208** [reply]
I'm facing the same issue with guzzlehttp/psr7 version problem when installing laravel-sitemap

**Reply by Tiththa** [reply]
I ended up fixing the issue by updating the composer `"guzzlehttp/guzzle": "^7.0.1"` and sitemap installed successfully.

**Comment by typoworx-de** [comment]
Is it possible to use laravel-sitemap without crawler using only manual "filling" with models using Sitemapable Interface? I also have the crawler-dependency issue having php8. I indeed want only model-filling at this point, but spatie/crawler is mandatory requirement.

**Reply by islandnuge** [reply]
Also hitting this with a fresh laravel project and php8.0.10 - following for updates!

**Reply by typoworx-de** [reply]
@frontalnugity as crawler is wired too deep into composer-requirements, I'm planning to do a soft-fork of this repo which will only support the 'base functionality' of sitemap management by using manual filling methods (no crawler). It has no absolute prio as I have some other urgent projects to be done. But in case you're already working on such a feature or you're interested in my (hopefully soon being done) solution, you're welcome to drop me a message. I'll drop an update here since I've got time to workout a soft-fork without crawler here.

**Reply by islandnuge** [reply]
@typoworx-de it's on my radar as an issue but also I have some competing priorities and was hoping for an update from spatie - if you do post an update I'm eager to take a peek!

**Comment by phuclh** [comment]
Any updates on it?

**Comment by mattvb91** [comment]
Still cant install this package:

```
    - spatie/image[1.5.3, ..., 1.7.6] require php ^7.0 -> your php version (8.1.9) does not satisfy that requirement.
```

**Reply by ousid** [reply]
Try to remove `composer.lock` and  `composer update`

**Comment by shahinaghamali** [comment]
do we have any solution for this issue?

**Comment by emrearpaci** [comment]
Delete composer.lock and try again.

**Comment by StayTick** [comment]
@amilmdr your answer worked for me. Thanks man
